server/session/security_cookie.ts

```diff
@@ -59,7 +59,7 @@ export function getSecurityCookieOptions(
       // TODO: with setting redirect attributes to support OIDC and SAML,
       //       we need to do additional cookie validation in AuthenticationHandlers.
       // if SAML fields present
-      if (sessionStorage.saml && sessionStorage.saml.requestId && sessionStorage.saml.nextUrl) {
```

Comment:
Can you explain why this is needed?

Reply:
After the change to remove `nextUrl` from the login flow on first login, the code started returning an error [here](https://github.com/opensearch-project/security-dashboards-plugin/blob/main/server/auth/types/saml/routes.ts#L115) and if you diff in it calls the `security_cookie.validate` flow which eventually hits this line. If you look [2 lines down](https://github.com/opensearch-project/security-dashboards-plugin/blob/main/server/auth/types/saml/routes.ts#L118-L120) it has logic to handle the case where nextUrl is empty and correctly redirects with the serverBasePath, but the or block was never being hit. 

The code to retain nextUrl for SAML was introduced in [this PR](https://github.com/opensearch-project/security-dashboards-plugin/pull/1039) and for OpenID was introduced in [this PR](https://github.com/opensearch-project/security-dashboards-plugin/pull/1563) using the same logic and both always pass `nextUrl` even on first login. It would also be fine to always pass `nextUrl` as long as it includes the basePath, but I opted to exclude it on initial login instead and have the login handlers redirect to the default route which includes the basePath. I opted to do this because we had existing code in place that would never get hit (The portion after `||` [here](https://github.com/opensearch-project/security-dashboards-plugin/blob/main/server/auth/types/saml/routes.ts#L118-L120) and [here](https://github.com/opensearch-project/security-dashboards-plugin/blob/main/server/auth/types/openid/routes.ts#L137)) and this change assures that it will be handled appropriately.